### PR TITLE
[SPARK-35364][PYTHON] Renaming the existing Koalas related codes

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

There are still naming related to Koalas in test and function name. This PR addressed them to fit pandas-on-spark.
- kdf -> psdf
- kser -> psser
- kidx -> psidx
- kmidx -> psmidx
- to_koalas() -> to_pandas_on_spark()

### Why are the changes needed?

This is because the name Koalas is no longer used in PySpark.

### Does this PR introduce _any_ user-facing change?

`to_koalas()` function is renamed to `to_pandas_on_spark()`

### How was this patch tested?

Tested in local manually.
After changing the related naming, I checked them one by one.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
@itholic can you rebase this? the linter failure was fixed in the upstream.

**Comment by HyukjinKwon** [comment]
cc @ueshin @xinrong-databricks FYI

**Comment by itholic** [comment]
Thanks, @HyukjinKwon . Just rebased

File: python/pyspark/pandas/tests/test_dataframe_conversion.py [hunk 1]
```diff
@@ -101,69 +101,69 @@ def test_to_excel(self):
             koalas_location = dirpath + "/" + "output2.xlsx"
```
Comment (xinrong-databricks) [inline python/pyspark/pandas/tests/test_dataframe_conversion.py:101]: Shall we rename `koalas_location`? Please ignore if it's out of the scope of this PR.

**Comment by xinrong-databricks** [comment]
python/pyspark/pandas/tests/test_utils.py has some variables named `koalas`. Shall we rename them as well? Or we may adjust them later in a separate PR.

**Review by xinrong-meng — APPROVED** [review]

**Review by ueshin — COMMENTED** [review]
What about `koalas` accessor? Maybe in the following PR?

File: python/pyspark/pandas/tests/test_dataframe_conversion.py (hunk 1 above)
Comment (ueshin) [reply python/pyspark/pandas/tests/test_dataframe_conversion.py:101]: Actually there are a few more places.
- `koalas_result` in `test_series.py`.
- a test name `test_as_spark_type_koalas_dtype` in `test_typedef.py`

File: python/pyspark/pandas/tests/test_indexing.py [hunk 2]
```diff
@@ -34,6 +34,10 @@ def pdf(self):
             {"month": [1, 4, 7, 10], "year": [2012, 2014, 2013, 2014], "sale": [55, 40, 84, 31]}
         )
 
+    @property
+    def psdf(self):
+        return ps.from_pandas(self.pdf)
+
```
[the remark this answers is not in the document]
Comment (itholic) [reply python/pyspark/pandas/tests/test_indexing.py:40]: Thanks!!

File: python/pyspark/pandas/tests/test_dataframe_conversion.py (hunk 1 above)
Comment (itholic) [reply python/pyspark/pandas/tests/test_dataframe_conversion.py]: Thanks!! Just addressed them

**Comment by itholic** [comment]
Thanks for the review, @ueshin , @xinrong-databricks !
> What about `koalas` accessor? Maybe in the following PR?
Yeah, I just created new JIRA for this : https://issues.apache.org/jira/browse/SPARK-35453
Let me fix this in the following PR.

File: python/pyspark/pandas/tests/indexes/test_base.py [hunk 3]
```diff
@@ -1224,104 +1230,106 @@ def test_monotonic(self):
...
-            kmidx = ps.from_pandas(pmidx)
-            self.assert_eq(kmidx.is_monotonic_increasing, False)
-            self.assert_eq(kmidx.is_monotonic_decreasing, False)
+            psmidx = ps.from_pandas(pmidx)
+            self.assert_eq(psmidx.is_monotonic_increasing, False)
+            self.assert_eq(psmidx.is_monotonic_decreasing, False)
 
         # Disable the test cases below because pandas returns `True` or `False` randomly.
         # else:
-            # [(-5, None), (-4, None), (-3, None), (-2, None), (-1, None)]
```
Comment (HyukjinKwon) [inline python/pyspark/pandas/tests/indexes/test_base.py:1252]: indentation here seems weird

File: python/pyspark/pandas/tests/indexes/test_datetime.py [hunk 4]
```diff
@@ -228,7 +228,8 @@ def test_arithmetic_op_exceptions(self):
 
     try:
         import xmlrunner  # type: ignore[import]
-        testRunner = xmlrunner.XMLTestRunner(output='target/test-reports', verbosity=2)
+
```
Comment (HyukjinKwon) [inline python/pyspark/pandas/tests/indexes/test_datetime.py:231]:
```suggestion
```

**Review by HyukjinKwon — APPROVED** [review]
LGTM

**Review by ueshin — APPROVED** [review]
LGTM.

**Comment by ueshin** [comment]
Thanks! merging to master.

**Comment by itholic** [comment]
Thanks for all the reviews :-)

File: python/pyspark/pandas/__init__.py [hunk 5]
```diff
@@ -144,7 +144,7 @@ def _auto_patch_spark():
 
         from pyspark.sql import dataframe as df
 
-        df.DataFrame.to_koalas = DataFrame.to_koalas
```
Comment (HyukjinKwon) [inline python/pyspark/pandas/__init__.py:147]: @itholic can we keep `to_koalas` for backward compatibility for now? We'll have to show a deprecated warning as well.

Comment (itholic) [reply python/pyspark/pandas/__init__.py:147]: Sounds good. Just created SPARK-35539 and working on it.

**Pull request closed without merging**